scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -856,14 +874,24 @@ object Execution {
    * these executions are executed in parallel: run is called on all at the
    * same time, not one after the other.
    */
+  private case class SequencingFn[T]() extends Function1[(T, List[T]), List[T]] {
+    def apply(results: (T, List[T])) = results match {
+      case (y, ys) => y :: ys
+    }
+  }
+  private case class ReversingFn[T]() extends Function1[List[T], List[T]] {
+    def apply(results: List[T]) = results.reverse
+  }
+  // Avoid recreating the empty Execution each time
+  private val nil = from(Nil)
   def sequence[T](exs: Seq[Execution[T]]): Execution[Seq[T]] = {
     @annotation.tailrec
     def go(xs: List[Execution[T]], acc: Execution[List[T]]): Execution[List[T]] = xs match {
       case Nil => acc
-      case h :: tail => go(tail, h.zip(acc).map { case (y, ys) => y :: ys })
+      case h :: tail => go(tail, h.zip(acc).map(SequencingFn()))
     }
     // This pushes all of them onto a list, and then reverse to keep order
-    go(exs.toList, from(Nil)).map(_.reverse)
+    go(exs.toList, nil).map(ReversingFn())
```

Comment:
a note, you could, like you have with `nil` make `val cons = { p: ((Any, List[Any])) => (p._1) :: (p_.2) }` then cast it to `Function1[(T, List[T]), T]` with `def seqFn[T] = cons.asInstanceOf[((T, List[T])) => List[T]]`. This won't fail on the jvm due to type erasure, and it will keep the functions referentially equivalent.

The same could be done with `reverse`